apps/src/templates/manageStudents/CodeReviewGroupsDialog.jsx

```diff
@@ -113,8 +113,7 @@ export default function CodeReviewGroupsDialog({
         setLoadingStatus(LOADING_STATES.LOADED);
       })
       .fail(() => setLoadingStatus(LOADING_STATES.ERROR));
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [dataApi]);
```

Comment:
I have to say I'm not sure I really agree with this linter hook. Calling `useEffect` with no dependencies can be a common idiom to cause some action such as an API call to occur exactly once at page load and then never again. Passing in dependencies changes this behavior, so that this code will be executed again whenever that dependency changes.

For example if one of these dependencies is a function, React will create a brand new function on every re-render by default. You can get around this by using [useCallback](https://react.dev/reference/react/useCallback), but you can see how this might be error-prone.

In this particular case the `dataApi` object is passed in as a `new CodeReviewGroupsDataApi(sectionId)` property when this component is included in the `ManageStudentsTable`. I _suspect_ that anytime this table is re-rendered, this API object will be reinstantiated and the `useEffect` call would therefore call the API all over again and reset the UI with a fresh copy of data from the server. In contrast, when you leave this out of the dependencies the API call only happens once on the first render and never again. Please check whether this is the case.

If this is a problem, a solution could involve instantiating this API object parameter outside of the `ManageStudentsTable.render` method.

Reply:
Noted and thank you for your comment. I think I was letting my emotions about getting this *small* tech debt ticket out of the way and didn't want to think about all the places the simple fix (to remove the linter errors) might not be the right one. Going to make a new ticket that reflects the complexity of this project and reach out to Bryan.

Reply:
https://codedotorg.atlassian.net/browse/TEACH-651